BrainPortal/cbrain_plugins/cbrain-plugins-base/cbrain_task/start_vm/common/start_vm.rb

```diff
@@ -0,0 +1,60 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2012
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.  
+#
+
+class CbrainTask::StartVM #:nodoc:
+  
+  def param_validation_errors #:nodoc:
+    errors = Hash.new
```

Comment:
The problem is that I'd like the same validation code to be called client- and bourreau-side. That's why it's in common. On the bourreau side, the params_error API is not available. So, I fill in my own hash and I use it in the portal-side after_form method to fill-in the params_errors object,  and in the bourreau-side setup method to add error messages in the task log. Does it make sense?